apps/src/code-studio/pd/workshop_dashboard/reports/report_table.jsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * Report Table
+ */
+import React from 'react';
+import _, {orderBy} from 'lodash';
+import {Table, sort} from 'reactabular';
+import wrappedSortable from '@cdo/apps/templates/tables/wrapped_sortable';
+import color from '@cdo/apps/color';
+
+const styles = {
+  container: {
+    overflowX: 'auto'
+  }
+};
+
+const ReportTable = React.createClass({
+  propTypes: {
+    columns: React.PropTypes.array.isRequired,
+    rows: React.PropTypes.array.isRequired
+  },
+
+  getInitialState() {
+    return {
+      sortingColumns: {
+        workshop_id: {
+          direction: 'asc',
+          position: 0
+        }
+      }
+    };
+  },
+
+  getSortingColumns() {
+    return this.state.sortingColumns || {};
+  },
+
+  onSort(selectedColumn) {
+    const sortingColumns = sort.byColumn({
+      sortingColumns: this.state.sortingColumns,
+      // Custom sortingOrder removes 'no-sort' from the cycle
+      sortingOrder: {
+        FIRST: 'asc',
+        asc: 'desc',
+        desc: 'asc'
+      },
+      selectedColumn
+    });
+
+    this.setState({
+      sortingColumns: sort.byColumn({
+        sortingColumns
+      })
+    });
+  },
+
+  getSortableTransform() {
+    return wrappedSortable(
+      this.getSortingColumns,
+      this.onSort,
+      {
+        container: {whiteSpace: 'nowrap'},
+        default: {color: color.light_gray}
+      }
+    );
+  },
+
+  addTransform(element, transform) {
+    element.transforms = element.transforms || [];
+    element.transforms.push(transform);
+  },
+
+  // Apply to all headers
+  applyHeaderTransforms(columns) {
+    const sortableTransform = this.getSortableTransform();
+    _.each(columns, column => {
+      this.addTransform(column.header, sortableTransform);
+    });
+
+    return columns;
+  },
+
+  render() {
+    // Since there may not be a unique id per row, add a rowKey based on pre-sorted index.
+    const rows = _.map(this.props.rows, (row, i) =>
+      _.merge(row, {
+        rowKey: i + 1
+      })
```

Comment:
you can even rewrite this as:
```
rows = {
  ...rows,
  rowKey: i + 1
}
```
if you're curious how this works, grep for `transform-object-rest-spread`